homeassistant/components/wolflink/sensor.py

```diff
@@ -0,0 +1,174 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up all entries for Wolf Platform."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+    parameters = hass.data[DOMAIN][config_entry.entry_id]["parameters"]
+
+    entities = []
+    for parameter in parameters:
+        if isinstance(parameter, Temperature):
+            entities.append(WolfLinkTemperature(coordinator, parameter))
+        if isinstance(parameter, Pressure):
+            entities.append(WolfLinkPressure(coordinator, parameter))
+        if isinstance(parameter, PercentageParameter):
+            entities.append(WolfLinkPercentage(coordinator, parameter))
+        if isinstance(parameter, ListItemParameter):
+            entities.append(WolfLinkState(coordinator, parameter))
+        if isinstance(parameter, HoursParameter):
+            entities.append(WolfLinkHours(coordinator, parameter))
+        if isinstance(parameter, SimpleParameter):
+            entities.append(WolfLinkSensor(coordinator, parameter))
+
+    async_add_entities(entities, True)
+
+
+class WolfLinkSensor(Entity):
+    """Base class for all Wolf entities."""
+
+    def __init__(self, coordinator, wolf_object: Parameter):
+        """Initialize."""
+        self.coordinator = coordinator
+        self.wolf_object = wolf_object
+
+    @property
+    def name(self):
+        """Return the name."""
+        return f"{self.wolf_object.name}"
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self.coordinator.data[self.wolf_object.value_id]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "parameter_id": self.wolf_object.parameter_id,
+            "value_id": self.wolf_object.value_id,
+            "parent": self.wolf_object.parent,
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return self.wolf_object.parameter_id
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return bool(self.state)
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self.coordinator.async_request_refresh()
+        _LOGGER.debug("Updating %s", self.coordinator.data[self.wolf_object.value_id])
+
+
+class WolfLinkHours(WolfLinkSensor):
+    """Class for hour based entities."""
+
+    @property
+    def icon(self):
+        """Icon to display in the front end."""
+        return "mdi:clock"
```

Comment:
We don't set a custom icon when we set device_class.

Reply:
Actually I am not sure if it is DEVICE_CLASS_TIMESTAMP. So I will not remove it.